js/views/userPageVw.js

```diff
@@ -255,40 +256,55 @@ module.exports = Backbone.View.extend({
       timeout: 4000,
       success: function(model, response){
         //don't render if view has been closed and the $el has been deleted
-        if(self.$el){
-          if (response.profile){
-            $('.js-loadingModal').addClass('hide');
-            if (self.options.ownPage === true){
-              model.set('headerURL', self.options.userModel.get('serverUrl') + "get_image?hash=" + model.get('profile').header_hash);
-              model.set('avatarURL', self.options.userModel.get('serverUrl') + "get_image?hash=" + model.get('profile').avatar_hash);
-            }else{
-              model.set('headerURL', self.options.userModel.get('serverUrl') + "get_image?hash=" + model.get('profile').header_hash + "&guid=" + self.pageID);
-              model.set('avatarURL', self.options.userModel.get('serverUrl') + "get_image?hash=" + model.get('profile').avatar_hash + "&guid=" + self.pageID);
-            }
+        if (self.isRemoved()) return;
```

Comment:
I've thought about that, but don't think it's a good idea.

For one, there's a whole slew of technical complications with trying to automagically add in some custom functionality to all server request handlers for a particular view.

Secondly, the current approach gives us a lot of flexibility in selectively allowing certain callback functionality to still happen, even if the view has been removed.. For example, on one of the POSTs in this view, I'm preventing any modal pop-ups to happen if the view has been removed, but I'm still allowing a fetch of the global user profile model to go through, since that affects other views, both present and future.